projects/igniteui-angular/core/src/core/i18n/formatters/formatter-intl.spec.ts

```diff
@@ -0,0 +1,92 @@
+        it('should format number correctly', () => {
+            expect(i18nFormatter.formatNumber(1.25, 'en')).toEqual('1.25');
+            expect(i18nFormatter.formatNumber(125, 'en')).toEqual('125');
+            expect(i18nFormatter.formatNumber(1250, 'en')).toEqual('1,250');
+            expect(i18nFormatter.formatNumber(12500, 'en')).toEqual('12,500');
+
+            expect(i18nFormatter.formatNumber(1.25, 'bg')).toEqual('1,25');
+            expect(i18nFormatter.formatNumber(125, 'bg')).toEqual('125');
+            expect(i18nFormatter.formatNumber(1250, 'bg')).toEqual('1250');
+            expect(i18nFormatter.formatNumber(12500, 'bg')).toEqual('12 500');
+
+            expect(i18nFormatter.formatNumber(1.25, 'de')).toEqual('1,25');
+            expect(i18nFormatter.formatNumber(125, 'de')).toEqual('125');
+            expect(i18nFormatter.formatNumber(1250, 'de')).toEqual('1.250');
+            expect(i18nFormatter.formatNumber(12500, 'de')).toEqual('12.500');
+        });
+
+        it('should format percent correctly', () => {
+            expect(i18nFormatter.formatPercent(1.25, 'en')).toEqual('125%');
+            expect(i18nFormatter.formatPercent(125, 'en')).toEqual('12,500%');
+            expect(i18nFormatter.formatPercent(1.25, 'bg')).toEqual('125%');
+            expect(i18nFormatter.formatPercent(125, 'bg')).toEqual('12 500%');
+        });
+
+        it('should format currency correctly', () => {
+            expect(i18nFormatter.formatCurrency(12345, 'en', 'symbol', 'EUR')).toEqual('€12,345.00');
+            expect(i18nFormatter.formatCurrency(12345, 'en', 'symbol', 'EUR', '1.0-3')).toEqual('€12,345');
+            expect(i18nFormatter.formatCurrency('12345.33', 'en', 'symbol', 'EUR', '1.0-3')).toEqual('€12,345.33');
+            expect(i18nFormatter.formatCurrency(12345, 'en', 'symbol', 'EUR', '1.1-3')).toEqual('€12,345.0');
+            expect(i18nFormatter.formatCurrency('12345', 'en', 'symbol', 'EUR', '1.1-3')).toEqual('€12,345.0');
+        });
+    })
+
+    describe('date formatting', () => {
+        it('should format string to dateTime using Intl', () => {
+            expect(i18nFormatter.formatDate('2025-01-25T14:15:00+02:00', 'short', 'en-US', "Europe/Sofia")).toEqual('1/25/25, 2:15 PM');
+            expect(i18nFormatter.formatDate('2025-01-25T14:15:00+02:00', 'medium', 'en-US', "Europe/Sofia")).toEqual('Jan 25, 2025, 2:15:00 PM');
+            expect(i18nFormatter.formatDate('2025-01-25T14:15:00+02:00', 'long', 'en-US', "Europe/Sofia")).toEqual('January 25, 2025 at 2:15:00 PM GMT+2');
+            expect(i18nFormatter.formatDate('2025-01-25T14:15:00+02:00', 'full', 'en-US', "Europe/Sofia")).toEqual('Saturday, January 25, 2025 at 2:15:00 PM Eastern European Standard Time');
+        });
+
+
+        it('should format string to time using Intl', () => {
+            expect(i18nFormatter.formatDate('2025-01-25T14:15:00+02:00', 'shortTime', 'en-US', "Europe/Sofia")).toEqual('2:15 PM');
```

Comment:
It might be better not to have the timezone offset hardcoded but maybe resolve it runtime. Since  the tests will probably fail on a machine in another timezone.

Reply:
The tests on the CI already run on another machine in different timezone. This test checks if you set it to explicitly different timezone, it should return the time in that timezone.